driver-sync/src/test/functional/com/mongodb/client/unified/RunOnRequirementsMatcher.java

```diff
@@ -21,37 +21,62 @@
 import org.bson.BsonDocument;
 import org.bson.BsonValue;
 
+import java.util.Map;
+
+import static com.mongodb.ClusterFixture.getServerParameters;
 import static com.mongodb.JsonTestServerVersionChecker.getMaxServerVersionForField;
-import static com.mongodb.JsonTestServerVersionChecker.getMinServerVersionForField;
+import static com.mongodb.JsonTestServerVersionChecker.getMinServerVersion;
 import static com.mongodb.JsonTestServerVersionChecker.topologyMatches;
 
 final class RunOnRequirementsMatcher {
     public static boolean runOnRequirementsMet(final BsonArray runOnRequirements, final ServerVersion serverVersion) {
         for (BsonValue cur : runOnRequirements) {
             boolean requirementMet = true;
             BsonDocument requirement = cur.asDocument();
-            if (requirement.containsKey("minServerVersion")
-                    && serverVersion.compareTo(getMinServerVersionForField("minServerVersion", requirement)) < 0) {
-                requirementMet = false;
-            }
-            if (requirement.containsKey("maxServerVersion")
-                    && serverVersion.compareTo(getMaxServerVersionForField("maxServerVersion", requirement)) > 0) {
-                requirementMet = false;
-            }
 
-            if (requirement.containsKey("topologies")) {
-                BsonArray topologyTypes = requirement.getArray("topologies");
-                if (!topologyMatches(topologyTypes)) {
-                    requirementMet = false;
+            requirementLoop:
+            for (Map.Entry<String, BsonValue> curRequirement : requirement.entrySet()) {
+                switch (curRequirement.getKey()) {
+                    case "minServerVersion":
+                        if (serverVersion.compareTo(getMinServerVersion(curRequirement.getValue().asString().getValue())) < 0) {
+                            requirementMet = false;
+                            break requirementLoop;
+                        }
+                        break;
+                    case "maxServerVersion":
+                        if (serverVersion.compareTo(getMaxServerVersionForField(curRequirement.getValue().asString().getValue())) > 0) {
+                            requirementMet = false;
+                            break requirementLoop;
+                        }
+                        break;
+                    case "topologies":
+                        BsonArray topologyTypes = curRequirement.getValue().asArray();
+                        if (!topologyMatches(topologyTypes)) {
+                            requirementMet = false;
+                            break requirementLoop;
+                        }
+                        break;
+                    case "serverParameters":
```

Comment:
With the addition of serverparameters supports, it's now necessary to break out of the loop as soon as a requirement is not met.  Otherwise we have to handle checking for a server parameter that's not supported by the server.